stock_picking_batch_actual_date/tests/test_stock_picking_batch_actual_date.py

```diff
@@ -0,0 +1,100 @@
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner = cls.env["res.partner"].create({"name": "Test Partner"})
+        product_category = cls.env["product.category"].create(
+            {
+                "name": "Test Category",
+                "property_cost_method": "fifo",
+                "property_valuation": "real_time",
+            }
+        )
+        cls.product_1 = cls.env["product.product"].create(
+            {
+                "name": "Test Product",
+                "type": "product",
+                "categ_id": product_category.id,
+                "standard_price": 100.0,
+            }
+        )
+        cls.supplier_location = cls.env.ref("stock.stock_location_suppliers")
+        cls.stock_location = cls.env.ref("stock.stock_location_stock")
+
+    def create_picking(self, actual_date=False):
+        receipt = self.env["stock.picking"].create(
+            {
+                "location_id": self.supplier_location.id,
+                "location_dest_id": self.stock_location.id,
+                "partner_id": self.partner.id,
+                "picking_type_id": self.env.ref("stock.picking_type_in").id,
+                "actual_date": actual_date,
+                "move_ids": [
+                    Command.create(
+                        {
+                            "name": "10 in",
+                            "location_id": self.supplier_location.id,
+                            "location_dest_id": self.stock_location.id,
+                            "product_id": self.product_1.id,
+                            "product_uom_qty": 10.0,
+                            "move_line_ids": [
+                                Command.create(
+                                    {
+                                        "product_id": self.product_1.id,
+                                        "location_id": self.supplier_location.id,
+                                        "location_dest_id": self.stock_location.id,
+                                        "qty_done": 10.0,
+                                    }
+                                )
+                            ],
+                        }
+                    )
+                ],
+            }
+        )
+        receipt.move_ids._action_confirm()
+        return receipt, receipt.move_ids
+
+    def test_stock_picking_batch_actual_date(self):
+        receipt_1, move_1 = self.create_picking()
+        receipt_2, move_2 = self.create_picking()
+        receipt_3, move_3 = self.create_picking()
+        batch_picking = self.env["stock.picking.batch"].create(
+            {
+                "actual_date": date(2025, 5, 5),
+                "picking_ids": [Command.set([receipt_1.id, receipt_2.id])],
+            }
+        )
+        self.assertEqual(batch_picking.actual_date, date(2025, 5, 5))
+        self.assertEqual(receipt_1.actual_date, date(2025, 5, 5))
+        self.assertEqual(move_1.actual_date, date(2025, 5, 5))
```

Comment:
We should leave the checks on stock move actual_date outside the scope of this test.
```suggestion
```